tests/foreman/virtwho/test_api_virtwho.py

```diff
@@ -107,3 +111,243 @@ def test_positive_deploy_configure_id(self):
+            vhd.hypervisor_id = value
+            vhd.update(['hypervisor_id'])
+            config_file = get_configure_file(vhd.id)
+            command = get_configure_command(vhd.id)
+            deploy_configure_by_command(command)
+            self.assertEqual(
+                get_configure_option('hypervisor_id', config_file),
+                value
+            )
+        vhd.delete()
+        self.assertEqual(entities.VirtWhoConfig().search(
+            query={'search': 'name={}'.format(name)}), [])
+
+    @tier1
+    def test_positive_filter_option(self):
+        """ Verify filter option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: 89cc1134-69d9-4da8-9ba9-a296c17f4f16
+
+        :expectedresults: filter and filter_hosts can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        whitelist = {
+            'filtering_mode': '1',
+            'whitelist': '.*redhat.com'
+        }
+        blacklist = {
+            'filtering_mode': '2',
+            'blacklist': '.*redhat.com'
+        }
+        # Update Whitelist and check the result
+        vhd.filtering_mode = whitelist['filtering_mode']
+        vhd.whitelist = whitelist['whitelist']
+        vhd.update(whitelist.keys())
+        config_file = get_configure_file(vhd.id)
+        command = get_configure_command(vhd.id)
+        deploy_configure_by_command(command)
+        self.assertEqual(
+            get_configure_option('filter_hosts', config_file),
+            whitelist['whitelist'])
+        # Update Blacklist and check the result
+        vhd.filtering_mode = blacklist['filtering_mode']
+        vhd.blacklist = blacklist['blacklist']
+        vhd.update(blacklist.keys())
+        config_file = get_configure_file(vhd.id)
+        command = get_configure_command(vhd.id)
+        deploy_configure_by_command(command)
+        self.assertEqual(
+            get_configure_option('exclude_hosts', config_file),
+            blacklist['blacklist'])
+        vhd.delete()
+        self.assertEqual(entities.VirtWhoConfig().search(
+            query={'search': 'name={}'.format(name)}), [])
+
+    @tier1
+    def test_positive_proxy_option(self):
+        """ Verify http_proxy option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id""
+
+        :id: 11352fee-5e00-4b24-9515-30a790685ede
+
+        :expectedresults: http_proxy and no_proxy option can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        proxy = 'test.rexample.com:3128'
+        no_proxy = 'test.satellite.com'
```

Comment:
Could you add a check point about the default `NO_PROXY=*` in /etc/sysconfig/virt-who?

Reply:
Sure, will add it with next commit